clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/client/AzureNetworkClient.groovy

```diff
@@ -51,12 +52,22 @@ class AzureNetworkClient extends AzureBaseClient {
     def result = new ArrayList<AzureLoadBalancerDescription>()
 
     try {
-      this.getNetworkResourceProviderClient(creds).getLoadBalancersOperations().listAll().getLoadBalancers().each {item ->
+      def loadBalancers = this.getNetworkResourceProviderClient(creds).getLoadBalancersOperations().listAll().getLoadBalancers()
```

Comment:
Is there a reason you are changing the code to get the list of resources into a local variable and then iterate over the local variable? I see that you've done this in a several places throughout the code and I'm curious why?